snuba/utils/streams/kafka.py

```diff
@@ -637,3 +645,71 @@ def close(self, timeout: Optional[float] = None) -> None:
         messages: int = self.__producer.flush(*[timeout] if timeout is not None else [])
         if messages > 0:
             raise TimeoutError(f"{messages} commit log messages pending delivery")
+
+
+class KafkaProducer(Producer[TPayload]):
+    def __init__(
+        self, configuration: Mapping[str, Any], codec: Codec[KafkaPayload, TPayload]
+    ) -> None:
+        self.__configuretion = configuration
+        self.__codec = codec
+
+        self.__producer = ConfluentProducer(configuration)
+        self.__shutdown_requested = Event()
+
+        self.__result = execute(self.__worker)
+
+    def __worker(self) -> None:
```

Comment:
Not sure why you need this worker at all. Is it just to ensure a flush is executed after the last message is accepted by the produce method and before closing the producer  ?

Reply:
`poll` (or `flush`) is the only thing that fires callbacks, so to use the `Future` interface, there has to be another thread that calls `poll` continuously, otherwise this example would deadlock:

```python
producer.produce(…).result()
```

Reply:
Added a description here: e2ce4acc036eb0f6d14a48cb780a558849dd7593